libs/ledger-live-common/src/families/cosmos/js-signOperation.ts

```diff
@@ -42,21 +45,44 @@ const signOperation: SignOperationFnSignature<Transaction> = ({ account, deviceI
           // Cosmos Nano App sign data in Amino way only, not Protobuf.
           // This is a legacy outdated standard and a long-term blocking point.
           const chainId = await cosmosAPI.getChainId();
-          const signDoc = makeSignDoc(
+          let signDoc = makeSignDoc(
             aminoMsgs,
             feeToEncode,
             chainId,
             transaction.memo || "",
             accountNumber.toString(),
             sequence.toString(),
           );
-          const ledgerSigner = new LedgerSigner(transport, {
-            hdPaths: [stringToPath("m/" + account.freshAddressPath)],
-            prefix: cryptoFactory(account.currency.id).prefix,
-          });
+          signDoc = sortObjectKeysDeeply(signDoc) as StdSignDoc;
+          const tx = Buffer.from(JSON.stringify(signDoc), "utf-8");
+
+          const app = new CosmosApp(transport);
+
+          const path = account.freshAddressPath.split("/").map(p => parseInt(p.replace("'", "")));
+
+          const { compressed_pk } = await app.getAddressAndPubKey(path, chainInstance.prefix);
+
+          const signResponseApp =
+            path[1] === 60
```

Comment:
Just for my curiosity: can you please explain why we need this condition?

Reply:
Done